indexer/packages/postgres/src/stores/pnl-ticks-table.ts

```diff
@@ -268,3 +290,159 @@ export async function findMostRecentPnlTickForEachAccount(
     'subaccountId',
   );
 }
+
+export async function getRankedPnlTicks(
+  timeSpan: string,
+): Promise<LeaderboardPnlCreateObject[]> {
+  if (timeSpan === 'ALL_TIME') {
+    return getLatestRankedPnlTicks();
+  }
+  return getRankedPnlTicksForTimeSpan(timeSpan);
+}
+
+async function getRankedPnlTicksForTimeSpan(
+  timeSpan: string,
+): Promise<LeaderboardPnlCreateObject[]> {
+  const vaultAddresses: string[] = getVaultAddresses();
+  const vaultAddressesString: string = vaultAddresses.map((address) => `'${address}'`).join(',');
+  const intervalSqlString: string = convertTimespanToSQL(timeSpan);
+  const result: {
+    rows: LeaderboardPnlCreateObject[]
+  } = await knexReadReplica.getConnection().raw(
+    `
+    WITH latest_subaccount_pnl_x_days_ago_ranked AS (
```

Comment:
WindowAgg  (cost=367986.26..367989.76 rows=200 width=148) (actual time=76309.323..76318.837 rows=16277 loops=1)
   ->  Sort  (cost=367986.26..367986.76 rows=200 width=108) (actual time=76309.312..76311.359 rows=16277 loops=1)
         Sort Key: aggregated_results."totalPnl" DESC
         Sort Method: quicksort  Memory: 2673kB
         ->  Subquery Scan on aggregated_results  (cost=367956.65..367978.62 rows=200 width=108) (actual time=76284.526..76303.321 rows=16277 loops=1)
               ->  GroupAggregate  (cost=367956.65..367976.62 rows=200 width=108) (actual time=76284.526..76301.609 rows=16277 loops=1)
                     Group Key: latest_pnl.address
                     ->  Sort  (cost=367956.65..367960.89 rows=1697 width=57) (actual time=76284.512..76286.680 rows=18127 loops=1)
                           Sort Key: latest_pnl.address
                           Sort Method: quicksort  Memory: 3135kB
                           ->  Subquery Scan on latest_pnl  (cost=367840.16..367865.61 rows=1697 width=57) (actual time=76139.205..76229.244 rows=18127 loops=1)
                                 ->  Unique  (cost=367840.16..367848.64 rows=1697 width=97) (actual time=76139.203..76226.909 rows=18127 loops=1)
                                       ->  Sort  (cost=367840.16..367844.40 rows=1697 width=97) (actual time=76139.202..76195.461 rows=325653 loops=1)
                                             Sort Key: a."subaccountId", a."blockHeight" DESC
                                             Sort Method: external merge  Disk: 29928kB
                                             ->  Nested Loop  (cost=0.57..367749.13 rows=1697 width=97) (actual time=11.067..75743.401 rows=325653 loops=1)
                                                   ->  Seq Scan on subaccounts b  (cost=0.00..674.17 rows=92 width=60) (actual time=0.010..11.217 rows=18128 loops=1)
                                                         Filter: (("subaccountNumber" % 128) = 0)
                                                         Rows Removed by Filter: 217
                                                   ->  Index Scan using pnl_ticks_subaccountid_createdat_index on pnl_ticks a  (cost=0.57..3989.71 rows=24 width=37) (actual time=4.152..4.174 rows=18 loops=18128)
                                                         Index Cond: ("subaccountId" = b.id)
                                                         Filter: (("createdAt")::date = CURRENT_DATE)
                                                         Rows Removed by Filter: 3651

Reply:
WindowAgg  (cost=780380.06..780383.56 rows=200 width=148) (actual time=266511.173..266520.779 rows=16277 loops=1)
   ->  Sort  (cost=780380.06..780380.56 rows=200 width=108) (actual time=266511.165..266512.302 rows=16277 loops=1)
         Sort Key: aggregated_results."totalPnl" DESC
         Sort Method: quicksort  Memory: 2673kB
         ->  Subquery Scan on aggregated_results  (cost=780346.20..780372.41 rows=200 width=108) (actual time=266486.377..266506.644 rows=16277 loops=1)
               ->  GroupAggregate  (cost=780346.20..780370.41 rows=200 width=108) (actual time=266486.376..266504.933 rows=16277 loops=1)
                     Group Key: b.address
                     ->  Sort  (cost=780346.20..780350.44 rows=1697 width=64) (actual time=266486.362..266488.777 rows=18127 loops=1)
                           Sort Key: b.address
                           Sort Method: quicksort  Memory: 3135kB
                           ->  Merge Left Join  (cost=779466.17..780255.17 rows=1697 width=64) (actual time=253355.362..266426.130 rows=18127 loops=1)
                                 Merge Cond: (a."subaccountId" = a_1."subaccountId")
                                 ->  Unique  (cost=367840.17..367848.66 rows=1697 width=97) (actual time=75718.199..75806.954 rows=18127 loops=1)
                                       ->  Sort  (cost=367840.17..367844.42 rows=1697 width=97) (actual time=75718.198..75779.857 rows=325653 loops=1)
                                             Sort Key: a."subaccountId", a."blockHeight" DESC
                                             Sort Method: external merge  Disk: 35032kB
                                             ->  Nested Loop  (cost=0.57..367749.14 rows=1697 width=97) (actual time=10.950..75394.181 rows=325653 loops=1)
                                                   ->  Seq Scan on subaccounts b  (cost=0.00..674.19 rows=92 width=60) (actual time=0.010..11.709 rows=18129 loops=1)
                                                         Filter: (("subaccountNumber" % 128) = 0)
                                                         Rows Removed by Filter: 217
                                                   ->  Index Scan using pnl_ticks_subaccountid_createdat_index on pnl_ticks a  (cost=0.57..3989.71 rows=24 width=37) (actual time=4.133..4.155 rows=18 loops=18129)
                                                         Index Cond: ("subaccountId" = b.id)
                                                         Filter: (("createdAt")::date = CURRENT_DATE)
                                                         Rows Removed by Filter: 3651
                                 ->  Unique  (cost=411625.99..412191.78 rows=14124 width=547) (actual time=177637.157..190608.495 rows=17529 loops=1)
                                       ->  Sort  (cost=411625.99..411908.88 rows=113157 width=547) (actual time=177637.155..186055.812 rows=63619443 loops=1)
                                             Sort Key: a_1."subaccountId", a_1."blockHeight" DESC
                                             Sort Method: external merge  Disk: 2987792kB
                                             ->  Nested Loop  (cost=0.57..370299.61 rows=113157 width=547) (actual time=0.032..97505.319 rows=63622388 loops=1)
                                                   ->  Seq Scan on subaccounts b_1  (cost=0.00..674.19 rows=92 width=16) (actual time=0.012..14.078 rows=18129 lo
ops=1)
                                                         Filter: (("subaccountNumber" % 128) = 0)
                                                         Rows Removed by Filter: 217
                                                   ->  Index Scan using pnl_ticks_subaccountid_createdat_index on pnl_ticks a_1  (cost=0.57..4001.69 rows=1598 wi
dth=31) (actual time=0.013..4.902 rows=3509 loops=18129)
                                                         Index Cond: ("subaccountId" = b_1.id)
                                                         Filter: (("createdAt")::date <= (CURRENT_DATE - '7 days'::interval))
                                                         Rows Removed by Filter: 159
 Planning Time: 0.601 ms
 Execution Time: 267091.460 ms

Reply:
first one is all time, second one is for 7 day timspan

Reply:
oddly testnet has more data than internal mainnet

Reply:
why doesnt the first one have an execution or planning time? I think what we really need is just how long each command takes, and you can use explain analyze to optimize it. In this case 263s seems on the long side, and im concerned that it could cause the read replica to lag